core/genesis.go

```diff
@@ -432,75 +435,16 @@ func LoadChainConfig(db ethdb.Database, genesis *Genesis) (*params.ChainConfig,
 	return params.BSCChainConfig, params.BSCGenesisHash, nil
 }
 
-// For any block or time in g.Config which is nil but the same field in defaultConfig is not
-// set the field in genesis config to the field in defaultConfig.
-// Reflection is used to avoid a long series of if statements with hardcoded block names.
-func (g *Genesis) setDefaultHardforkValues(defaultConfig *params.ChainConfig) {
-	// Regex to match Block names or Time names
-	hardforkPattern := []string{`.*Block$`, `.*Time$`}
-
-	for _, pat := range hardforkPattern {
-		hardforkRegex := regexp.MustCompile(pat)
-
-		// Get reflect values
-		gConfigElem := reflect.ValueOf(g.Config).Elem()
-		defaultConfigElem := reflect.ValueOf(defaultConfig).Elem()
-
-		// Iterate over fields in config
-		for i := 0; i < gConfigElem.NumField(); i++ {
-			gConfigField := gConfigElem.Field(i)
-			defaultConfigField := defaultConfigElem.Field(i)
-			fieldName := gConfigElem.Type().Field(i).Name
-
-			// Use the regex to check if the field is a Block or Time field
-			if gConfigField.Kind() == reflect.Ptr && hardforkRegex.MatchString(fieldName) {
-				if gConfigField.IsNil() {
-					gConfigField.Set(defaultConfigField)
-				}
-			}
-		}
-	}
-}
-
-// Hard fork field specified in config.toml has higher priority, but
-// if it is not specified in config.toml, use the default height in code.
 func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
-	var defaultConfig *params.ChainConfig
+	conf := params.GetBuiltInChainConfig(ghash)
 	switch {
```

Comment:
how about using if statement here? maybe clearer.
```go
conf := params.GetBuiltInChainConfig(ghash)
if conf != nil {
    return conf
}
if g != nil {
    return g.Config // it could be a custom config for QA test, just return
}
return params.AllEthashProtocolChanges
```

Reply:
> how about using if statement here? maybe clearer.
> 
> ```go
> conf := params.GetBuiltInChainConfig(ghash)
> if conf != nil {
>     return conf
> }
> if g != nil {
>     return g.Config // it could be a custom config for QA test, just return
> }
> return params.AllEthashProtocolChanges
> ```

accepted